src/sentry/static/sentry/app/utils/queryString.jsx

```diff
@@ -4,10 +4,20 @@ export function formatQueryString(qs) {
 }
 
 // returns environment name from query or null if not specified
-// Any charater can be valid in an environment name
+// Any character can be valid in an environment name but we need to
+// check for matching environments with the quotation marks first
+// to match the way tag searches are being done
 export function getQueryEnvironment(qs) {
-  const match = qs.match(/environment:([^\s]*)/);
-  return match ? match[1] : null;
+  const matchWithQuotes = qs.match(/environment:"([^\s]*)"/);
```

Comment:
I was going to say that you can do it in 1 regex e.g. `/environment:("|)([^\s]*)\1/`, but we probably want to allow spaces when quotes are used. I'm sure there's a way to do this in 1 regex (cc self-proclaimed regex god @benvinegar) but having 2 will allow us lesser mortals to understand it better.

Also, if we allow quotes to be used, then how do we allow searching for literal quotes, etc. We should start defining what we want to allow in our test cases.

Reply:
Maybe allow a set of delimiters?

```
'"`|/\
```

Reply:
Updated to support spaces